dev/Repeater/UniformGridLayout.cpp

```diff
@@ -213,10 +213,11 @@ winrt::Rect UniformGridLayout::Algorithm_GetExtent(
 
     if (itemsCount > 0)
     {
+        // Only use all of the space if item stretch is fill, otherwise size layout according to items placed
         extent.*MinorSize() =
-            std::isfinite(availableSizeMinor) ?
+            std::isfinite(availableSizeMinor) && m_itemsStretch == winrt::UniformGridLayoutItemsStretch::Fill ?
             availableSizeMinor :
-            std::max(0.0f, itemsCount * GetMinorSizeWithSpacing(context) - static_cast<float>(MinItemSpacing()));
```

Comment:
FlowLayout likely has the same issue as well.

Reply:
I just looked at FlowLayout(Algorithm).cpp and it didn't seem like that the same issue was also present there.